DQM/GEM/plugins/GEMDAQStatusSource.cc

```diff
@@ -27,36 +27,36 @@ void GEMDAQStatusSource::fillDescriptions(edm::ConfigurationDescriptions &descri
 }
 
 void GEMDAQStatusSource::LoadROMap(edm::EventSetup const &iSetup) {
-  auto gemROMap = std::make_shared<GEMROMapping>();
   //if (useDBEMap_)
   if (true) {
-    const auto &eMap = iSetup.getData(gemEMapToken_);
-    auto gemEMap = std::make_unique<GEMeMap>(eMap);
-    gemEMap->convert(*gemROMap);
-
-    for (auto imap : gemEMap->theChamberMap_) {
-      int nNumChamber = (int)imap.fedId.size();
-      for (int i = 0; i < nNumChamber; i++) {
-        unsigned int fedId = imap.fedId[i];
-        uint8_t amcNum = imap.amcNum[i];
-        uint8_t gebId = imap.gebId[i];
-        GEMROMapping::chamEC geb_ec{fedId, amcNum, gebId};
-        GEMROMapping::chamDC geb_dc = gemROMap->chamberPos(geb_ec);
-        GEMDetId gemChId = geb_dc.detId;
-
-        mapFEDIdToRe_[fedId] = gemChId.region();
-        mapAMC13ToListChamber_[fedId].push_back(gemChId);
-        mapAMCToListChamber_[{fedId, amcNum}].push_back(gemChId);
-      }
+    const auto &chMap = iSetup.getData(gemChMapToken_);
+    auto gemChMap = std::make_unique<GEMChMap>(chMap);
+
+    for (auto const &[ec, dc] : gemChMap->chamberMap()) {
+      unsigned int fedId = ec.fedId;
+      uint8_t amcNum = ec.amcNum;
+      GEMDetId gemChId(dc.detId);
+
+      mapFEDIdToRe_[fedId] = gemChId.region();
+      mapAMC13ToListChamber_[fedId].push_back(gemChId);
+      mapAMCToListChamber_[{fedId, amcNum}].push_back(gemChId);
     }
 
-    gemEMap.reset();
   } else {
     // no EMap in DB, using dummy
     // FIXME: How to add mapFEDIdToRe_ and mapDetIdToAMC_??
```

Comment:
please update the comment, now you actually access `mapFEDIdToRe_` and `mapDetIdToAMC_`